lib/zipkin-tracer/zipkin_rabbit_mq_sender.rb

```diff
@@ -0,0 +1,42 @@
+require 'json'
+require 'zipkin-tracer/zipkin_sender_base'
+require 'zipkin-tracer/hostname_resolver'
+
+module Trace
+  class RabbitMqPublisher
+    def initialize(connection)
+      @connection = connection
+      @channel = @connection.create_channel
+    end
+
+    def publish(exchange, routing_key, message)
+      exchange = @channel.exchange(exchange)
+      exchange.publish(message, routing_key: routing_key)
```

Comment:
If this is a slow operation the thread on your application working on this will stop till it is done.
Consider , if it makes sense, to make it async (you can copy from how json sender is doing it)

Reply:
I rewrote the publisher using suckerpunch to make it async similar to the other senders.